constants/keys.py

```diff
@@ -4,14 +4,32 @@
 from enum import auto
 
 
-class Tasks(StrEnum):
+class JobServices(StrEnum):
+    ADAPT = "Adapt"
+    KOMPRESS = "Kompress"
+
+
+class KompressTasks(StrEnum):
     TEXT_GENERATION = "llm"
     CLASSIFICATION = "image_classification"
 
 
+class AdaptTasks(StrEnum):
+    TEXT_GENERATION = "text_generation"
+    TEXT_CLASSIFICATION = "text_classification"
+    SEQ_2_SEQ_TRANSLATION = "Seq2Seq_tasks"
+    SEQ_2_SEQ_SUMMARIZATION = "Seq2Seq_tasks"
+    QUESTION_ANSWERING = "question_answering"
+    IMAGE_CLASSIFICATION = "image_classification"
+    OBJECT_DETECTION = "object_detection"
+    IMAGE_SEGMENTATION = "image_segmentation"
+    POSE_ESTIMATION = "pose_estimation"
+
+
 class FactoryTypes(StrEnum):
-    TEXT_GENERATION: Tasks = Tasks.TEXT_GENERATION
-    VISION: Tasks = Tasks.CLASSIFICATION  # default
+    TEXT_GENERATION: KompressTasks = KompressTasks.TEXT_GENERATION
+    VISION: KompressTasks = KompressTasks.CLASSIFICATION  # default
+    ADAPT: JobServices = JobServices.ADAPT
```

Comment:
This is exactl what I was saying from the beginning. Having both Tasks and Job_service is inconsistent. But since, it was decided to have it this way when I was making those changes, I think it can stay for now. Moreover the more logical change is to modify komperess to have separation based on job_service rather than task I think and that should require another PR.

Reply:
> Moreover the more logical change is to modify kompress to have separation based on job_service rather than task

What are these changes that you suggest? can u elaborate more

Reply:
Before that, could you elaborate o what exactly you meant by this :

> since we now have all three components; we can have an auto enum and avoid referencing tasks/job services. besides, using tasks and job services is inconsistent.

Reply:
yes. creating different enums for Tasks -> TextGen, Vision, Adapt and JobService -> Kompress, Adapt. And using those in factory methods and logics. (without using a task enum in job enum, or vice-versa)